src/handshakestate.rs

```diff
@@ -264,6 +290,117 @@ impl HandshakeState {
         Ok(byte_index)
     }
 
+    #[cfg(feature = "hfs")]
+    fn _write_message_token(&mut self, message: &mut [u8], token: Token) -> Result<usize, Error> {
+        match token {
+            Token::E => self._write_message_token_e(message),
+            Token::S => self._write_message_token_s(message),
+            Token::Psk(n) => self._write_message_token_psk(n),
+            Token::Dhee => self._write_message_token_dh(false, false),
+            Token::Dhes => self._write_message_token_dh(false, true),
+            Token::Dhse => self._write_message_token_dh(true, false),
+            Token::Dhss => self._write_message_token_dh(true, true),
+            Token::E1 => self._write_message_token_e1(message),
+            Token::Ekem1 => self._write_message_token_ekem1(message),
```

Comment:
If possible, if we could also use the method I suggested in the error `fmt()` code to reduce duplicate code here, that would be best.